common/src/test/scala/uk/ac/wellcome/sqs/SQSReaderTest.scala

```diff
@@ -15,44 +15,81 @@ class SQSReaderTest
     with IntegrationPatience
     with SQSLocal {
 
+  override def queueName: String = "id_minter_queue"
+
   it("should get messages from the SQS queue, limited by the maximum number of messages and return them") {
     val sqsConfig =
-      SQSConfig("eu-west-1",
-                idMinterQueueUrl,
-                waitTime = 20 seconds,
-                maxMessages = 2)
+      SQSConfig("eu-west-1", queueUrl, waitTime = 20 seconds, maxMessages = 2)
     val messageStrings = List("someMessage1", "someMessage2", "someMessage3")
-    messageStrings.foreach(sqsClient.sendMessage(idMinterQueueUrl, _))
+    messageStrings.foreach(sqsClient.sendMessage(queueUrl, _))
     val sqsReader =
       new SQSReader(sqsClient, sqsConfig)
 
-    val futureMessages = sqsReader.retrieveMessages()
+    var receivedMessages: List[Message] = Nil
 
-    whenReady(futureMessages) { messages =>
+    val futureMessages = sqsReader.retrieveAndDeleteMessages(message => receivedMessages = message :: receivedMessages)
+
+    whenReady(futureMessages) { _ =>
       // SQS is not a FIFO queue and it only guarantees that a message is sent at least once,
       // not that it is received exactly once
-      messages should have size 2
-      messages.foreach { message =>
+      receivedMessages should have size 2
+      receivedMessages.foreach { message =>
         messageStrings should contain(message.getBody)
       }
```

Comment:
Now in fact we can go back to a slightly stronger assertion: the two messages should be _distinct_, because we delete them from the queue after processing.